web/controllers/userController/index.ts

```diff
@@ -1007,7 +1011,58 @@ async function putActive(wix_id: string, active: boolean, person: Pupil | Studen
                 await dissolveMatch(matches[i], 0, person);
             }
 
-            // Step 2: Deactivate
+            // Step 2: Cancel all courses if user is student
+            if (type == Student) {
+                let courses = await getConnection()
+                    .getRepository(Course)
+                    .createQueryBuilder("course")
+                    .leftJoinAndSelect("course.instructors", "instructors")
+                    .where("instructors.id = :id", { id: person.id})
+                    .getMany();
+
+                console.log("CRS: ", courses);
+
+                entityManager.transaction(async em => {
+                    for (const course of courses) {
+                        if (!course.instructors.some(i => i.id === person.id)) { // Only proceed if we're part of this course as an instructor
+                            return;
```

Comment:
Ah, my comment was slightly missleading here, in a `.forEach`, `return` returns control to the forEach logic and continues with the next entry, in a `for of` you'd rather need a `continue;` here ...